cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -296,6 +299,48 @@ def create_self_config(ca, client):
     build_kubeconfig(server)
 
 
+@when('ceph-storage.available')
```

Comment:
I don't see a `@when_not` preventing this method from running more than once. I see a lot of file manipulation in this method that may not be idempotent.  Am I wrong?